js/core/template_manager.js

```diff
@@ -0,0 +1,221 @@
+
+        return !onlyJunkTemplateContent
+            ? { template: this._createTemplate($anonymousTemplate), name: this._anonymousTemplateName }
+            : {};
+    }
+
+    _createTemplateIfNeeded(templateSource) {
+        const cachedTemplate = this._tempTemplates.filter(tempTemplate =>
+            tempTemplate.source === templateKey(templateSource)
+        )[0];
+        if(cachedTemplate) return cachedTemplate.template;
+
+        const template = this._createTemplate(templateSource);
+        this._tempTemplates.push({ template, source: templateKey(templateSource) });
+        return template;
+    }
+
+    _createTemplate(templateSource) {
+        return this._createElement(validateTemplateSource(templateSource));
+    }
+
+    getTemplate(templateSource, templates, { isAsyncTemplate, skipTemplates }, context) {
+        if(!isFunction(templateSource)) {
+            return this._acquireTemplate(templateSource, this._createTemplateIfNeeded, templates, isAsyncTemplate, skipTemplates);
+        }
+
+        return new FunctionTemplate((options) => {
+            const templateSourceResult = templateSource.apply(context, getNormalizedTemplateArgs(options));
+
+            if(!isDefined(templateSourceResult)) {
+                return new EmptyTemplate();
+            }
+
+            let dispose = false;
+            const template = this._acquireTemplate(templateSourceResult, (templateSource) => {
+                if(templateSource.nodeType || isRenderer(templateSource) && !$(templateSource).is('script')) {
+                    return new FunctionTemplate(() => templateSource);
+                }
+                dispose = true;
+                return this._createTemplate(templateSource);
+            }, templates, isAsyncTemplate, skipTemplates);
+
+            const result = template.render(options);
+            dispose && template.dispose && template.dispose();
+            return result;
+        });
+    }
+
+    _acquireTemplate(templateSource, createTemplate, templates, isAsyncTemplate, skipTemplates) {
+        if(templateSource == null) {
+            return new EmptyTemplate();
+        }
+
+        if(templateSource instanceof ChildDefaultTemplate) {
+            return this._defaultTemplates[templateSource.name];
+        }
+
+        if(templateSource instanceof TemplateBase) {
+            return templateSource;
+        }
+
+        // TODO: templateSource.render is needed for angular2 integration. Try to remove it after supporting TypeScript modules.
+        if(isFunction(templateSource.render) && !isRenderer(templateSource)) {
+            return addOneRenderedCall(templateSource);
+        }
+
+        if(templateSource.nodeType || isRenderer(templateSource)) {
+            return createTemplate($(templateSource));
+        }
+
+        return this._acquireStringTemplate(templateSource, createTemplate, templates, isAsyncTemplate, skipTemplates);
```

Comment:
What about this?
```suggestion
        return this._acquireIntegrationTemplate(templateSource, templates, isAsyncTemplate, skipTemplates)
            || this._defaultTemplates[templateSource]
            || createTemplate(templateSource);
```
One arg less